webpack/module-minifier-plugin/src/AsyncImportCompressionPlugin.ts

```diff
@@ -0,0 +1,277 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import webpack, { Compiler, Plugin } from 'webpack';
+import { ReplaceSource } from 'webpack-sources';
+import { Tapable, TapOptions } from 'tapable';
+
+import Template from 'webpack/lib/Template';
+
+import { STAGE_AFTER } from './Constants';
+import {
+  IExtendedModule,
+  IModuleMinifierPluginHooks,
+  IPostProcessFragmentContext
+} from './ModuleMinifierPlugin.types';
+
+const PLUGIN_NAME: 'AsyncImportCompressionPlugin' = 'AsyncImportCompressionPlugin';
+
+const TAP_AFTER: TapOptions<'sync'> = {
+  name: PLUGIN_NAME,
+  stage: STAGE_AFTER
+};
+
+const ASYNC_IMPORT_PREFIX: '__IMPORT_ASYNC' = '__IMPORT_ASYNC';
+const ASYNC_IMPORT_REGEX: RegExp = /__IMPORT_ASYNC[^\)]+/g;
+
+declare class WebpackImportDependency extends webpack.compilation.Dependency {
+  public module: webpack.compilation.Module;
+  public block: {
+    chunkGroup: webpack.compilation.ChunkGroup;
+    range: [number, number];
+  };
+}
+
+interface IAsyncImportMetadata {
+  chunkCount: number;
+  chunkIds: number[];
+  count: number;
+  index: number;
+}
+
+interface ILocalImportMetadata {
+  meta: IAsyncImportMetadata;
+  module: webpack.compilation.Module;
+}
+
+function getImportDependency(compilation: webpack.compilation.Compilation): typeof WebpackImportDependency {
+  for (const key of compilation.dependencyTemplates.keys()) {
+    if (key.name === 'ImportDependency') {
+      return key as unknown as typeof WebpackImportDependency;
+    }
+  }
+
+  throw new Error(`Could not find ImportDependency!`);
+}
+
+function getImportTypeExpression(
+  module: webpack.compilation.Module,
+  originModule: webpack.compilation.Module
+): string {
+  const exportsType: string | undefined = module.buildMeta?.exportsType;
+  const strict: boolean | undefined = originModule.buildMeta?.strictHarmonyModule;
+
+  if (exportsType === 'namespace') {
+    return '';
+  } else if (exportsType === 'named') {
+    return ',3';
```

Comment:
Where do these numbers come from? Can you stick them in consts with meaningful names?

Reply:
They come from webpack's source code. I'll add some comments or at least a reference.